keyboards/planck/keymaps/muzfuz/keymap.c

```diff
@@ -0,0 +1,404 @@
+#ifdef KEYBOARD_planck_rev5
+      PORTE |= (1 << 6);
+#endif
+    }
+    return false;
+    break;
+  case PLOVER:
+    if (record->event.pressed)
+    {
+#ifdef AUDIO_ENABLE
+      stop_all_notes();
+      PLAY_SONG(plover_song);
+#endif
+      layer_off(_RAISE);
+      layer_off(_LOWER);
+      layer_off(_ADJUST);
+      layer_on(_PLOVER);
+      if (!eeconfig_is_enabled())
+      {
+        eeconfig_init();
+      }
+      keymap_config.raw = eeconfig_read_keymap();
+      keymap_config.nkro = 1;
+      eeconfig_update_keymap(keymap_config.raw);
+    }
+    return false;
+    break;
+  case EXT_PLV:
+    if (record->event.pressed)
+    {
+#ifdef AUDIO_ENABLE
+      PLAY_SONG(plover_gb_song);
+#endif
+      layer_off(_PLOVER);
+    }
+    return false;
+    break;
+  }
+  return true;
+}
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+void encoder_update(bool clockwise)
+{
+  if (muse_mode)
+  {
+    if (IS_LAYER_ON(_RAISE))
+    {
+      if (clockwise)
+      {
+        muse_offset++;
+      }
+      else
+      {
+        muse_offset--;
+      }
+    }
+    else
+    {
+      if (clockwise)
+      {
+        muse_tempo += 1;
+      }
+      else
+      {
+        muse_tempo -= 1;
+      }
+    }
+  }
+  else
+  {
+    if (clockwise)
+    {
+#ifdef MOUSEKEY_ENABLE
+      register_code(KC_MS_WH_DOWN);
+      unregister_code(KC_MS_WH_DOWN);
+#else
+      register_code(KC_PGDN);
+      unregister_code(KC_PGDN);
+#endif
+    }
+    else
+    {
+#ifdef MOUSEKEY_ENABLE
+      register_code(KC_MS_WH_UP);
+      unregister_code(KC_MS_WH_UP);
+#else
+      register_code(KC_PGUP);
+      unregister_code(KC_PGUP);
+#endif
+    }
+  }
+}
+
+void dip_update(uint8_t index, bool active)
+{
+  switch (index)
+  {
+  case 0:
+    if (active)
+    {
+#ifdef AUDIO_ENABLE
+      PLAY_SONG(plover_song);
+#endif
+      layer_on(_ADJUST);
+    }
+    else
+    {
+#ifdef AUDIO_ENABLE
+      PLAY_SONG(plover_gb_song);
+#endif
+      layer_off(_ADJUST);
+    }
+    break;
+  case 1:
+    if (active)
+    {
+      muse_mode = true;
+    }
+    else
+    {
+      muse_mode = false;
+#ifdef AUDIO_ENABLE
+      stop_all_notes();
+#endif
+    }
+  }
+}
+
+void matrix_scan_user(void)
+{
+#ifdef AUDIO_ENABLE
+  if (muse_mode)
+  {
+    if (muse_counter == 0)
+    {
+      uint8_t muse_note = muse_offset + SCALE[muse_clock_pulse()];
+      if (muse_note != last_muse_note)
+      {
+        stop_note(compute_freq_for_midi_note(last_muse_note));
+        play_note(compute_freq_for_midi_note(muse_note), 0xF);
+        last_muse_note = muse_note;
+      }
+    }
+    muse_counter = (muse_counter + 1) % muse_tempo;
+  }
+#endif
+}
+
+bool music_mask_user(uint16_t keycode)
+{
+  switch (keycode)
+  {
+  case RAISE:
+  case LOWER:
+    return false;
+  default:
+    return true;
+  }
+}
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```

Comment:
Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.